src/App.js

```diff
@@ -3,29 +3,51 @@ import React, { useState, useEffect } from "react";
 import "./App.css";
 
 function App() {
-    const [messages, setMessages] = useState(["hello", "I love GX!", "where am i????"]);
-    const [newMessage, setNewMessage] = useState("");
+    const [posts, setPosts] = useState([["hello", "I love GX!", "#0fa000","cum"],["hlo", "I hate GX!", "#b7ac59","cum"]]);
+    const [message, setNewMessage] = useState("");
+	const [name, setNewName] = useState("");
+	const [color, setNewColor] = useState("#0fa000");
+	
+	
 
-    const handleNewMsgChange = (e) => {
+    
+	const handleNewMsgChange = (e) => {
         setNewMessage(e.target.value);
     };
+	
+	const handleNameChange = (e) => {
+		setNewName(e.target.value);	
+	}; 
+	
+	const handleColorChange = (e) => {
+		setNewColor(e.target.value);
+	};
+	
     const handleNewMsgSubmit = (e) => {
         e.preventDefault();
-        setMessages([...messages, newMessage]);
+        setPosts([...posts, [message,name,color]]);
         setNewMessage("");
     };
+
+	
+	
     return (
         <div className="App">
             <img src="/gx-logo.jpg" style={{ position: "sticky", top: "5px", left: "5px", height: "50px" }} />
             <div id="chatbox-parent">
                 <div id="chatbox">
-                    {messages.map((msg) => (
-                        <div className="message">{msg}</div>
+                    {posts.map((msg) => (
+                        <div className="post">
+							<p className="name" style={{color:msg[2]}}>{msg[1]}</p>
```

Comment:
And then here we'd do (e.g.) `msg.color`, `msg.name` rather than `msg[2]`, `msg[1]`